plasmapy/analysis/time_series/conditional_averaging.py

```diff
@@ -0,0 +1,402 @@
+            conditional_events
+        )
+
+        self._peaks = signal[peak_indices]
+        self._number_of_events = len(self._peaks)
+
+        self._arrival_times = time[peak_indices]
+        self._waiting_times = np.diff(self._arrival_times)
+
+        if self._astropy_signal_unit is not None:
+            self._peaks *= self._astropy_signal_unit
+            self._conditional_average *= self._astropy_signal_unit
+
+        if self._astropy_time_unit is not None:
+            self._return_time *= self._astropy_time_unit
+            self._arrival_times *= self._astropy_time_unit
+            self._waiting_times *= self._astropy_time_unit
+
+    @property
+    def time(self):
+        """
+        Time values corresponding to the analysis window.
+
+        Returns
+        -------
+        time : 1D |array_like|
+            Time values representing the analysis window.
+        """
+        return self._return_time
+
+    @property
+    def average(self):
+        """
+        Conditional average over events.
+
+        Returns
+        -------
+        average : 1D |array_like|
+            Array representing the conditional average over events.
+
+        """
+        return self._conditional_average
+
+    @property
+    def variance(self):
+        """
+        Conditional variance over events.
+
+        Returns
+        -------
+        variance : 1D |array_like|
+            Array representing the conditional variance over events.
+
+        """
+        return self._conditional_variance
+
+    @property
+    def peaks(self):
+        """
+        Peak values of conditional events.
+
+        Returns
+        -------
+        peaks : 1D |array_like|
+            Peak values of conditional events.
+
+        """
+        return self._peaks
+
+    @property
+    def waiting_times(self):
+        """
+        Waiting times between consecutive peaks.
+
+        Returns
+        -------
+        waiting_times : 1D |array_like|
+            Waiting times between consecutive peaks of conditional events.
+
+        """
+        return self._waiting_times
+
+    @property
+    def arrival_times(self):
+        """
+        Arrival times corresponding to the conditional events.
+
+        Returns
+        -------
+        arrival_times : 1D |array_like|
+            Arrival times the conditional events.
+
+        """
+        return self._arrival_times
+
+    @property
+    def number_of_events(self):
+        """
+        Total number of conditional events.
+
+        Returns
+        -------
+        number_of_events : int
+            Total number of conditional events.
+
+        """
+        return self._number_of_events
+
+    def _check_for_value_errors(
+        self,
+        distance,
+        signal,
+        time,
+        reference_signal,
+        length_of_return,
+        upper_threshold,
+        lower_threshold,
+    ):
+        if distance < 0:
+            raise ValueError("distance can't be negative")
```

Comment:
Capitalise the start of the error messages please

Reply:
`distance` is a parameter starting with a lower case d, should it still be capitalized in the error message?

Reply:
Good point!  In that case, perhaps we could put distance in single quotes?